### PR TITLE
Fix duration parsing of .mkv and .webm files

### PR DESCRIPTION
**Pull request by Crydsch**
This fixes the duration parsing of some .mkv and .webm files.
The problem was that there is no "duration" property in the streams of the ffprobe json output.
But there is one within the "tags" property .

I changed the millisecond stripping to remove additional ticks as well.

### CODE REVIEW

File: FFMpegCore/FFMPEG/FFProbe.cs [hunk 1]
```diff
@@ -76,9 +76,20 @@ private VideoInfo ParseVideoInfoInternal(VideoInfo info, string probeOutput)
...
+            {
+                duration = TimeSpan.FromSeconds(double.TryParse(sDuration, NumberStyles.Any, CultureInfo.InvariantCulture, out var output) ? output : 0);
+            }
+            else
+            {
+                sDuration = (video ?? audio).Tags.Duration;
+                TimeSpan.TryParse(sDuration.Remove(sDuration.LastIndexOf('.')), CultureInfo.InvariantCulture, out duration);
+            }
+            // Strip milliseconds and additional ticks
+            info.Duration = new TimeSpan(duration.Days, duration.Hours, duration.Minutes, duration.Seconds);
```
Comment (rosenbjerg) [inline FFMpegCore/FFMPEG/FFProbe.cs]: Why not just `info.Duration = duration`?

Comment (rosenbjerg) [reply FFMpegCore/FFMPEG/FFProbe.cs]: Is it necessary to strip anything from the timespan?

Comment (Crydsch) [reply FFMpegCore/FFMPEG/FFProbe.cs]: I wanted to keep the same granularity as in the original code (seconds), but we could of course increase the resolution and use the 'whole' TimeSpan?

Comment (rosenbjerg) [reply FFMpegCore/FFMPEG/FFProbe.cs]: I think using the 'whole' TimeSpan would be better. The rest is just a matter of representation then :)

Comment (Crydsch) [reply FFMpegCore/FFMPEG/FFProbe.cs]: Ok, short update; i changed it to `info.Duration = duration` and some tests failed
Interesting enough there are strange double rounding issues:
In the Video_Duration test:
Does not work
`Assert.AreEqual(video.Duration.TotalSeconds - 5, outputVideo.Duration.TotalSeconds, Double.Epsilon);`
Does work
```
Assert.AreEqual(video.Duration.Hours, outputVideo.Duration.Hours);
Assert.AreEqual(video.Duration.Minutes, outputVideo.Duration.Minutes);
Assert.AreEqual(video.Duration.Seconds - 5, outputVideo.Duration.Seconds);
Assert.AreEqual(video.Duration.Milliseconds, outputVideo.Duration.Milliseconds);
```

Maybe there are some Ticks off - but i am not sure if we can expect an encoded video to be millisecond or even tick perfekt

Comment (Crydsch) [reply FFMpegCore/FFMPEG/FFProbe.cs]: Should we only check up to seconds?

File: FFMpegCore/FFMPEG/FFProbe.cs [hunk 2]
```diff
@@ -85,10 +85,9 @@ private VideoInfo ParseVideoInfoInternal(VideoInfo info, string probeOutput)
             else
             {
                 sDuration = (video ?? audio).Tags.Duration;
-                TimeSpan.TryParse(sDuration.Remove(sDuration.LastIndexOf('.')), CultureInfo.InvariantCulture, out duration);
+                TimeSpan.TryParse(sDuration.Remove(sDuration.LastIndexOf('.') + 8), CultureInfo.InvariantCulture, out duration); // TimeSpan fractions only allow up to 7 digits
             }
-            // Strip milliseconds and additional ticks
-            info.Duration = new TimeSpan(duration.Days, duration.Hours, duration.Minutes, duration.Seconds);
+            info.Duration = duration;
```
Comment (Crydsch) [inline FFMpegCore/FFMPEG/FFProbe.cs:91]: Increased duration resolution and updated the regarding tests.
I think it only makes sense to test the resulting duration up to seconds.
Individual tick differences are to be expected when converting a video.

File: FFMpegCore/FFMPEG/FFProbe.cs (hunk 1 above)
Comment (rosenbjerg) [reply FFMpegCore/FFMPEG/FFProbe.cs]: Yeah, I think that would make good sense.
As you say, we cannot expect them to be tick perfect

**Comment by rosenbjerg** [comment]
I have fixed to CI and verified that the unit tests still pass, and have now merged your PR. Thanks for the fix!

**Pull request merged**